lib/srv/forward/sshserver.go

```diff
@@ -636,8 +637,34 @@ func (s *Server) newRemoteClient(ctx context.Context, systemLogin string) (*trac
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-
 	return client, nil
+
+}
+
+// signersWithSHA1Fallback returns the signers provided by signersCb and appends
+// the same signers with forced SHA-1 signature to the end. This behavior is
+// required as older OpenSSH version <= 7.6 don't support SHA-2 signed certificates.
+func signersWithSHA1Fallback(signersCb func() ([]ssh.Signer, error)) func() ([]ssh.Signer, error) {
```

Comment:
nit: I don't have strong opinion but for me the function callback funcations are hard to track so wonder if we can consider calling signersCb before entering the `signersWithSHA1Fallback` function. 

```suggestion
func signersWithSHA1Fallback(signers []ssh.Signer) func() ([]ssh.Signer, error) {
```
```go
signers, err  := ssh.PublicKeysCallback(signersWithSHA1Fallback(s.userAgent.Signers))
if err != nil {
	return nil, trace.Wrap(err)
}
authMethod := ssh.PublicKeysCallback(signersWithSHA1Fallback(signers))
```